test/test_um2netcdf.py

```diff
@@ -651,6 +662,379 @@ def test_fix_units_do_nothing_no_um_units(ua_plev_cube):
+    lat_points = np.arange(-90 + 0.5*D_LAT_N96, 90.,
+                           D_LAT_N96, dtype="float32")
+
+    lat_standard_eg_coord = iris.coords.DimCoord(
+        points=lat_points,
+        standard_name=um2nc.LATITUDE
+    )
+    return lat_standard_eg_coord
+
+
+@pytest.fixture
+def lon_standard_eg():
+    # iris DimCoord imitating the standard longitude
+    # grid from CM2 (which uses the End Game grid).
+    lon_points = np.arange(0.5*D_LON_N96, 360, D_LON_N96, dtype="float32")
+    lon_standard_eg_coord = iris.coords.DimCoord(
+        points=lon_points,
+        standard_name=um2nc.LONGITUDE
+    )
+    return lon_standard_eg_coord
+
+
+class DummyCubeWithCoords(DummyCube):
+    # DummyCube with coordinates, which can be filled with
+    # iris.DimCoord objects for testing.
+    def __init__(self, dummy_cube, coords=[]):
+        super().__init__(dummy_cube.item_code,
+                         dummy_cube.var_name,
+                         dummy_cube.attributes,
+                         dummy_cube.units)
+
+        # Set up coordinate dictionary to hold each dummy coordinate, with keys
+        # given by the coordinate names. This ensures that the key used to
+        # access a coordinate via the coord method matches the
+        # coordinate's name.
+        self.coordinate_dict = {}
+        for coord in coords:
+            self.coordinate_dict[coord.name()] = coord
+
+    def coord(self, coordinate_name):
+        return self.coordinate_dict[coordinate_name]
+
+
+def assert_unmodified_coordinates(lat_coord, lon_coord):
+    """
+    Helper function to check that a coordinate's attributes match
+    those expected for a coordinate that has not yet been modified
+    by fix_latlon_coords.
+    """
+    for coord in [lat_coord, lon_coord]:
+        assert coord.points.dtype == np.dtype("float32")
+        assert not coord.has_bounds()
+        assert coord.var_name is None
+
+
+def assert_dtype_float64(lat_coord, lon_coord):
+    assert lat_coord.points.dtype == np.dtype("float64")
+    assert lon_coord.points.dtype == np.dtype("float64")
+
+
+def assert_has_bounds(lat_coord, lon_coord):
+    assert lat_coord.has_bounds()
+    assert lon_coord.has_bounds()
+
+
+# Tests of fix_latlon_coords. This function converts coordinate points
+# to double, adds bounds, and adds var_names to the coordinates.
+# The following tests check that these are done correctly.
+def test_fix_latlon_coords_river(ua_plev_cube,
+                                 lat_river,
+                                 lon_river):
+    """
+    Tests of the fix_lat_lon_coords function on river grid coordinates.
+    """
+
+    cube_with_river_coords = DummyCubeWithCoords(
+        dummy_cube=ua_plev_cube,
+        coords=[lat_river, lon_river]
+    )
+
+    # Supplementary grid information used by fix_latlon_coords.
+    # Grid type doesn't matter for river grid, but must be set.
+    grid_type = um2nc.GRID_END_GAME
```

Comment:
Minor style thing, `grid_type` can be replaced & the constant used directly (+ reduce the test slightly)